File: app_java/src/main/java/tech/thdev/android_mvp_sample/adapter/contract/ImageAdapterContract.java
package tech.thdev.android_mvp_sample.adapter.contract;

import java.util.ArrayList;

import tech.thdev.android_mvp_sample.data.ImageItem;
import tech.thdev.android_mvp_sample.listener.OnItemClickListener;

/**
 * Created by tae-hwan on 12/27/16.
 */
public interface ImageAdapterContract {

    interface View {

        void setOnClickListener(OnItemClickListener clickListener);

        void notifyAdapter();
    }

    interface Model {

        void addItems(ArrayList<ImageItem> imageItems);

        void clearItem();

        ImageItem getItem(int position);
    }
}

File: app_kotlin/src/main/java/tech/thdev/app_kotlin/adapter/contract/ImageAdapterContract.kt
package tech.thdev.app_kotlin.adapter.contract

import tech.thdev.app_kotlin.data.ImageItem
import java.util.*

/**
 * Created by tae-hwan on 12/27/16.
 */
interface ImageAdapterContract {

    interface View {

        var onClickFunc : ((Int) -> Unit)?

        fun notifyAdapter()
    }

    interface Model {

        fun addItems(imageItems: ArrayList<ImageItem>)

        fun clearItem()

        fun getItem(position: Int): ImageItem
    }
}